### ISSUE TITLE
Change activity to context when accessing application context

### ISSUE THREAD

**Issue by radzish** [tekartik/sqflite#53]
When plugin is been registered activity is used:

```
    public static void registerWith(Registrar registrar) {
        final MethodChannel channel = new MethodChannel(registrar.messenger(), "com.tekartik.sqflite");
        channel.setMethodCallHandler(new SqflitePlugin(registrar.activity().getApplicationContext()));
    }
```

would be great to use registrar's context instead:

```
    public static void registerWith(Registrar registrar) {
        final MethodChannel channel = new MethodChannel(registrar.messenger(), "com.tekartik.sqflite");
        channel.setMethodCallHandler(new SqflitePlugin(registrar.context().getApplicationContext()));
    }
```

This is needed to support "headless" mode, when activity not necessarily available.

**Comment by alextekartik** [comment]
Thanks for the report. I have to admit that I used whatever the template did without much care. I'll fix that. Any way to test its effect? (i.e. how to test the headless mode)

**Comment by alextekartik** [comment]
Fixed in 0.8.7 published on pub

**Comment by radzish** [comment]
thanks for quick fix! testing headless mode is quite complicated but you can use https://pub.dartlang.org/packages/android_alarm_manager package to create alarm that will wake up the app and execute some code with no activity instantiated.
